packages/flutter/lib/src/rendering/shifted_box.dart

```diff
@@ -98,20 +98,22 @@ class RenderPadding extends RenderShiftedBox {
        assert(padding.isNonNegative),
        _textDirection = textDirection,
        _padding = padding,
-       super(child) {
-    _applyUpdate();
-  }
+       super(child);
 
-  // The resolved absolute insets.
+  bool _needsResolution = true;
   EdgeInsets _resolvedPadding;
```

Comment:
Same thought here about using the nullity of _resolvedPadding to represent that the value needs to be computed.